homeassistant/components/neato/vacuum.py

```diff
@@ -241,14 +241,32 @@ def update(self):
             and self._robot_maps[self._robot_serial]
         ):
             allmaps = self._robot_maps[self._robot_serial]
+            _LOGGER.debug("Found the following maps for '%s': %s", self._name, allmaps)
+            self._robot_boundaries = {}  # Reset boundaries before refreshing boundaries
             for maps in allmaps:
                 try:
-                    self._robot_boundaries = self.robot.get_map_boundaries(
-                        maps["id"]
-                    ).json()
+                    robot_boundaries = self.robot.get_map_boundaries(maps["id"]).json()
                 except NeatoRobotException as ex:
                     _LOGGER.error("Could not fetch map boundaries: %s", ex)
-                    self._robot_boundaries = {}
+                    return
+
+                _LOGGER.debug(
+                    "Boundaries for robot '%s' in map '%s': %s",
+                    self._name,
+                    maps["name"],
+                    robot_boundaries,
+                )
+                if self._robot_boundaries == {}:
+                    self._robot_boundaries = robot_boundaries
+                else:
+                    self._robot_boundaries["data"]["boundaries"] += robot_boundaries[
```

Comment:
Except `robot_boundaries["data"]["boundaries"]` we ignore all elements of the second/third/... `robot_boundary` object. Is this intended?

Reply:
Yes so if we have multiple floor plans we need to add them to the same list so they can be properly identified and ran with.  Previously the variable was being overwritten: https://community.home-assistant.io/t/issue-with-neato-botvac-zones/167722/38?u=dshokouhi

You can see in this users statement the last debug we expect to see both sets of boundaries to exist but they do not.  So what I do is if its the first map we replace it completely and during the loop any additional maps that are there we just grab the boundaries and add it to our existing JSON list.

Reply:
This was with the old code of course I just added some debug statements to see what the variable was returning and I noticed that it was never saving all of them across all maps.  All the data we really need is in `boundaries`.

Reply:
So why not make it like this:

```python
self.boundaries = []
for map in allmaps:
    robot_boundaries = self.robot.get_map_boundaries(maps["id"]).json()
    self.boundaries += _robot_boundaries["data"]["boundaries"]
```

It's a bit simplified, but I think you get my point. This way no one ever asks for other attributes ob a `robot_boundaries` object. Since we ignore most of them, they seem not that important anyways.

Reply:
From the view of `single_responsibility` it's a bit cleaner this way

Reply:
We would also need to adjust line 398 too because it depends on this location :)  I can update this in a future PR when I add additional fixes.

Reply:
I'm okay with that. But would it hurt to change this right now?

Reply:
Yea let me test it :)

Reply:
Tested and submitted, thank you for the suggestion!